ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/hooks/getTableUIState.ts

```diff
@@ -0,0 +1,80 @@
+import { SearchFilter } from 'types/search';
+
+type IdleState = {
+    type: 'IDLE';
+};
+
+type LoadingState = {
+    type: 'LOADING';
+};
```

Comment:
I do still think it would be useful to distinguish the initial load from a refetch, at least if the UX ends up being different between the two states. To me, swapping out for a loading spinner e.g. on each page change is a bit jarring visually.

Reply:
Yeah, that's worth discussing. I'll highlight this as a point in the doc so UX can give their input as well

Reply:
Yes, please distinguish. Some pages like clusters and violations refetch every 5 seconds.

Reply:
[The Doc](https://docs.google.com/document/d/1biNKn8QWvlWVHE2lO8lGFvmJ-moJbuvm2mWT3akQZgw/edit) should account for **POLLING**, if that's what we're mentioning here. Did we want a distinction separate for that too?

Reply:
I was thinking that we have these cases:
1. The data is being explicitly loaded (initial load, search filters changed, paginating, user triggers a manual re-fetch)
2. The data is being re-fetched (polling)

For (1) we just consider all of that as **LOADING** and show a spinner. It indicates that the data is changing because of a user-driven decision. 

For (2) we consider that as **POLLING** and the UX is TBD (waiting on UX for their input). All I know is that we would want the UX to be non-intrusive. The user would want to know that the data is changing but also not so in their face that it affects their user experience.

Reply:
I'll ping UX about suggestions about differentiating between the initial load and the other types of loading.